ntoskrnl/config/cmalloc.c

```diff
@@ -176,6 +176,9 @@ CmpAllocateKeyControlBlock(VOID)
             /* Now go back and search the list */
             goto SearchKcbList;
         }
+
+        /* Release the allocation lock */
+        KeReleaseGuardedMutex(&CmpAllocBucketLock);
```

Comment:
Should a `return NULL;` too be added, similar to [line 260](https://github.com/reactos/reactos/blob/008571755a54a859a26353fd1bf30daae05b1592/ntoskrnl/config/cmalloc.c#L260)?

Reply:
Not really. If the allocation page call failed, the code falls back to the behaviour down below, of making a pool allocation for just one single KCB. (And if that allocation fails as well, you can see that the code will return a NULL value.)

Reply:
Thanks for confirming the fallback is actually wanted.